### ISSUE TITLE
Update Xcode 16 Beta 6 to Xcode 16 RC

### ISSUE THREAD

**Issue by jeslyvarghese** [actions/runner-images#10583]
### Tool name

Xcode

### Tool license

Proprietary with open-source components

### Add or update?

- [ ] Add
- [X] Update

### Desired version

16 RC

### Platforms where you need the tool

- [ ] Azure DevOps
- [ ] GitHub Actions

### Runner images where you need the tool

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] Ubuntu 24.04
- [ ] macOS 12
- [ ] macOS 13
- [ ] macOS 13 Arm64
- [ ] macOS 14
- [X] macOS 14 Arm64
- [ ] Windows Server 2019
- [ ] Windows Server 2022

**Comment by jeslyvarghese** [comment]
Requesting for an early update here to help developers who are planning to do AppStore submissions with latest iOS version on day 1 of iOS 18 release.

**Comment by sureshe456** [comment]
Hi @jeslyvarghese - We are looking into this issue. will keep you posted updates.

**Comment by jonasman** [comment]
When can we expect the deployment?

**Comment by sureshe456** [comment]
Hi @jonasman - We are planning to start the deployment end of next week.

**Comment by jeslyvarghese** [comment]
@sureshe456 Really unfortunate Github can't get something to production fast enough! Makes our release pipeline built around GitHub action unusable for AppStore submission until then.

**Comment by enricenrich** [comment]
> We are planning to start the deployment end of next week.

@sureshe456, is there any way to speed up the deployment and do it this week (ASAP)? The release of iOS 18 and iPadOS 18 is next Monday, so we need to be able to use Xcode 16 RC this week (maximum on Friday to ensure we are on time).

**Comment by bfreiband** [comment]
> > We are planning to start the deployment end of next week.
> 
> 
> 
> @sureshe456, is there any way to speed up the deployment and do it this week (ASAP)? The release of iOS 18 and iPadOS 18 is next Monday, so we need to be able to use Xcode 16 RC this week (maximum on Friday to ensure we are on time).

Plusplus to this, for iOS dev Mac runners it's really helpful and important to have a tight turnaround on at least the important new Xcode versions

**Comment by jeslyvarghese** [comment]
> > We are planning to start the deployment end of next week.
> 
> @sureshe456, is there any way to speed up the deployment and do it this week (ASAP)? The release of iOS 18 and iPadOS 18 is next Monday, so we need to be able to use Xcode 16 RC this week (maximum on Friday to ensure we are on time).

+1, faster availability is super important for us Apple developers to be ready for day 1 on iOS 18. Anything that can speed up this process will be a life saver!

**Comment by JUSTINMKAUFMAN** [comment]
+1

**Comment by BrentMifsud** [comment]
Why the long delay for the release version? All the betas were added pretty quickly.

**Comment by sarathrajsrinivasan** [comment]
Hi everyone,

We have started the deployment for Xcode 16 RC for macOS14-arm64. Will update here once the VM rollout is done.
Should be available before Monday EOD.

**Comment by jonasman** [comment]
> Hi @jonasman - We are planning to start the deployment end of next week.

Unfortunately this is not acceptable to us. 
When the RC comes out, we need to publish the new iOS 18 code quickly to be ready for the release one week later. Waiting 2 weeks for your deployment will completely destroy the changes to be ready on day 1 using github automations.

If this same happens when Xcode 16 is released, then it would mean we would need to wait yet another 2 weeks to be able to use our automations. So in total there would be 4 weeks that we need to ignore our deployment automations, and im not even counting the time when a new intermediate Xcode comes out like 16.1.

I urge you to review this practices because this is not acceptable for a fast paced company.

**Comment by jeslyvarghese** [comment]
[Bitrise](https://stacks.bitrise.io/changelogs/osx-xcode-16.0.x-edge/) and [CircleCI](https://discuss.circleci.com/t/xcode-16-1-beta-1-released/51693) seems to be more mindful around  developer needs here.

**Comment by enricenrich** [comment]
Based on https://github.com/actions/runner-images/pull/10603, Xcode 16 RC for `macOS14-arm64` is now available. I've just tested it.

**Comment by brentpi** [comment]
@sarathrajsrinivasan What about `macos-14` (not arm64) for Azure Pipelines?  Is this planned for a PR?

**Comment by sarathrajsrinivasan** [comment]
Hi everyone,
Xcode 16 RC is now available for macOS14-arm64. Please check.

@brentpi ,
We have started the deployment of macOS-14 for the current week. Should be available by Wednesday EOD.

**Comment by NachoSoto** [comment]
Which Xcode version is it using? Apple says it's not valid:

```

Error: Error uploading IPA to TestFlight:
{
    "tool-version": "8.003.16003",
    "tool-path": "/Applications/Xcode_16_Release_Candidate.app/Contents/SharedFrameworks/ContentDeliveryServices.framework/Versions/A/Frameworks/AppStoreService.framework",
    "os-version": "14.6.1",
    "product-errors": [
        {
            "message": "Asset validation failed",
            "userInfo": {
                "NSUnderlyingError": "Error Domain=IrisAPI Code=-19241 \"Asset validation failed\" UserInfo={status=409, detail=Unsupported SDK or Xcode version. Your app was built with an SDK or version of Xcode that isn\u2019t supported. Although you can use beta versions of SDKs and Xcode to build and upload apps to App Store Connect, you need to use the latest Release Candidates (RC) for SDKs and Xcode to submit the app. For details on currently supported SDKs and versions of Xcode, visit: https://developer.apple.com/news/releases., id=1d45c704-3387-47b0-99c7-fb1f2c2640ea, code=STATE_ERROR.VALIDATION_ERROR.90534, title=Asset validation failed, NSLocalizedFailureReason=Unsupported SDK or Xcode version. Your app was built with an SDK or version of Xcode that isn\u2019t supported. Although you can use beta versions of SDKs and Xcode to build and upload apps to App Store Connect, you need to use the latest Release Candidates (RC) for SDKs and Xcode to submit the app. For details on currently supported SDKs and versions of Xcode, visit: https://developer.apple.com/news/releases., NSLocalizedDescription=Asset validation failed}",
                "NSLocalizedDescription": "Asset validation failed",
                "iris-code": "STATE_ERROR.VALIDATION_ERROR.90534",
                "NSLocalizedFailureReason": "Unsupported SDK or Xcode version. Your app was built with an SDK or version of Xcode that isn\u2019t supported. Although you can use beta versions of SDKs and Xcode to build and upload apps to App Store Connect, you need to use the latest Release Candidates (RC) for SDKs and Xcode to submit the app. For details on currently supported SDKs and versions of Xcode, visit: https://developer.apple.com/news/releases. (ID: 1d45c704-3387-47b0-99c7-fb1f2c2640ea)"
            },
```

**Comment by enricenrich** [comment]
@NachoSoto I've opened a new issue to update to the latest RC: https://github.com/actions/runner-images/issues/10629

**Comment by NachoSoto** [comment]
Thanks. Do you know what the timeframe is to have that ready?

**Comment by enricenrich** [comment]
@NachoSoto, I have no idea. I've requested the upgrade a few hours ago and they mentioned they are looking into this. It might take a few days... 😔

For reference, the first RC was released by Apple on September 9th and it only became available on GitHub Actions on September 15th.
